js/state/frameStateUtil.js

```diff
@@ -33,27 +33,31 @@ const matchFrame = (queryInfo, frame) => {
 }
 
 function query (state, queryInfo) {
-  return state.get('frames').filter(matchFrame.bind(null, queryInfo))
+  return getFrames(state).filter(matchFrame.bind(null, queryInfo))
 }
 
 function find (state, queryInfo) {
-  return state.get('frames').find(matchFrame.bind(null, queryInfo))
+  return getFrames(state).find(matchFrame.bind(null, queryInfo))
 }
 
 function isFrameKeyActive (state, frameKey) {
   return getActiveFrameKey(state) === frameKey
 }
 
 function getFrames (state) {
-  return state.get('frames')
+  return state
+    ? state.get('frames')
+      ? state.get('frames').filter((frame) => !!frame)
```

Comment:
Is this going to be expensive calling this iterative filter function many many times each time we render / check state for changes? That will be fixed with #12105, but in a different release, but just a little concerned this will affect performance as in a render-cycle, possibly hundreds of frameStateUtil functions will get called, and each one will then run this filter.